### ISSUE TITLE
add support for common name matching 

### ISSUE THREAD

**Issue by jhpoelen** [globalbioticinteractions/nomer#27]
common names are . . . commonly used (commonly, common names are used for vertebrates). And curated lists exists relating common names to their scientific counterpart. 

Suggest to add support to match again these common name to scientific name lists.

Wilson, D. E., and F. R. Cole. 2000. Common Names of Mammals of the World. Smithsonian Institution Press, Washington, D.C.

**Comment by jhpoelen** [comment]
fyi @seltmann

**Comment by jhpoelen** [comment]
e.g., chicken -> Gallus gallus .

**Comment by seltmann** [comment]
David Bloom and John Wieczorek suggested several possible sources for curated common names of vertebrates. Two sources for birds - Avibase and EBird.  For amphibians, AmphibiaWeb. For all EOL.

Animal Diversity Web at U of Michigan. Their UI is set up to accept both scientific and common names, but this may just be a keyword function like VertNet/GBIF.iDigBio. Given the fact that their intended audience is more public and less research-oriented, they might indeed have a mapping - their search results also suggest that they might have one with both scientific and common names displayed for each result (see meadowlark - https://animaldiversity.org/search/?q=meadowlark&feature=INFORMATION).

**Comment by jhpoelen** [comment]
NCBI Taxonomy and ITIS also have common names. Is this something to consider?

**Comment by jhpoelen** [comment]
here's some chickens from NCBI - 

```
$ cat names.dmp  | grep "common name" | grep chicken
5630	|	chicken-of-the-woods	|		|	common name	|
9004	|	greater prairie chicken	|		|	genbank common name	|
9031	|	chicken	|		|	genbank common name	|
9031	|	chickens	|		|	common name	|
29588	|	golden chicken fern	|		|	common name	|
109042	|	lesser prairie-chicken	|		|	genbank common name	|
115121	|	Attwater's prairie chicken	|		|	genbank common name	|
171959	|	chicken-corn	|		|	common name	|
186210	|	brown chicken louse	|		|	genbank common name	|
186210	|	reddish-brown chicken louse	|		|	common name	|
186211	|	giant chicken louse	|		|	genbank common name	|
186211	|	large chicken louse	|		|	common name	|
206448	|	chicken-claws	|		|	common name	|
220712	|	chicken liver sponge	|		|	genbank common name	|
241518	|	hen and chickens fern	|		|	common name	|
310867	|	chickenthief	|		|	common name	|
318061	|	chickenweed	|		|	common name	|
360507	|	chicken grunt	|		|	genbank common name	|
375434	|	mountain chicken	|		|	common name	|
375434	|	mountain chicken frog	|		|	genbank common name	|
```

**Comment by jhpoelen** [comment]
and vernaculars (common names) from ITIS - 

```
$ curl --silent "https://zenodo.org/record/3833105/files/vernaculars.gz" | gunzip | grep chicken
20432|chickenweed|English|N|03/29/2012|145598
28397|chicken tree|English|N|02/27/2012|77229
34608|chickenspike|English|N|09/26/2012|151320
38429|chickensage|English|N|10/05/2010|117369
38430|silver chickensage|English|N|10/05/2010|117370
38431|gray chickensage|English|N|10/05/2010|117371
38433|separateleaf chickensage|English|N|10/05/2010|117374
38434|fivefinger chickensage|English|N|10/05/2010|117375
38434|cinquefoil chickensage|unspecified|N|10/05/2010|75073
38435|Zion chickensage|English|N|10/05/2010|117379
38436|Laramie chickensage|English|N|10/05/2010|117380
42108|chicken corn|unspecified|N|12/01/2010|77020
151455|chicken dung fly|unspecified|Y|05/09/2003|9475
503240|hen and chickens|English|N|05/31/2012|146370
503792|chickenthief|English|N|06/27/2012|148712
505012|chickenclaws|English|N|08/30/2011|128119
505323|compact chickensage|English|N|10/05/2010|117373
530509|fivefinger chickensage|English|N|10/05/2010|117376
530509|alkali chickensage|unspecified|N|10/05/2010|75075
530510|fivefinger chickensage|English|N|10/05/2010|117377
530510|false chicken-sage|English|N|10/05/2010|117378
551216|chicken sprat|English|N|12/19/2002|59474
```

**Comment by jhpoelen** [comment]
initial support for matching common names via [ncbi-taxon] matcher has been added :

```
$ echo -e "\tchicken" | nomer append ncbi-taxon
using matcher [ncbi-taxon]
NCBI taxonomy already indexed at [/media/jorrit/data/nomer/ncbi/ncbi], no need to import.
	chicken	COMMON_NAME_OF	NCBI:9031	Gallus gallus	species		root | cellular organisms | Eukaryota | Opisthokonta | Metazoa | Eumetazoa | Bilateria | Deuterostomia | Chordata | Craniata | Vertebrata | Gnathostomata | Teleostomi | Euteleostomi | Sarcopterygii | Dipnotetrapodomorpha | Tetrapoda | Amniota | Sauropsida | Sauria | Archelosauria | Archosauria | Dinosauria | Saurischia | Theropoda | Coelurosauria | Aves | Neognathae | Galloanserae | Galliformes | Phasianidae | Phasianinae | Gallus | Gallus gallus	NCBI:1 | NCBI:131567 | NCBI:2759 | NCBI:33154 | NCBI:33208 | NCBI:6072 | NCBI:33213 | NCBI:33511 | NCBI:7711 | NCBI:89593 | NCBI:7742 | NCBI:7776 | NCBI:117570 | NCBI:117571 | NCBI:8287 | NCBI:1338369 | NCBI:32523 | NCBI:32524 | NCBI:8457 | NCBI:32561 | NCBI:1329799 | NCBI:8492 | NCBI:436486 | NCBI:436489 | NCBI:436491 | NCBI:436492 | NCBI:8782 | NCBI:8825 | NCBI:1549675 | NCBI:8976 | NCBI:9005 | NCBI:9072 | NCBI:9030 | NCBI:9031	|  | superkingdom | clade | kingdom | clade | clade | clade | phylum | subphylum | clade | clade | clade | clade | superclass | clade | clade | clade | clade | clade | clade | clade | clade | clade | clade | clade | class | infraclass | superorder | order | family | subfamily | genus | species	https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=9031	
```

ncbi knows about dogs too:

```
$ echo -e "\tdog" | nomer append ncbi-taxon
using matcher [ncbi-taxon]
NCBI taxonomy already indexed at [/media/jorrit/data/nomer/ncbi/ncbi], no need to import.
	dog	COMMON_NAME_OF	NCBI:9615	Canis lupus familiaris	subspecies		root | cellular organisms | Eukaryota | Opisthokonta | Metazoa | Eumetazoa | Bilateria | Deuterostomia | Chordata | Craniata | Vertebrata | Gnathostomata | Teleostomi | Euteleostomi | Sarcopterygii | Dipnotetrapodomorpha | Tetrapoda | Amniota | Mammalia | Theria | Eutheria | Boreoeutheria | Laurasiatheria | Carnivora | Caniformia | Canidae | Canis | Canis lupus | Canis lupus familiaris	NCBI:1 | NCBI:131567 | NCBI:2759 | NCBI:33154 | NCBI:33208 | NCBI:6072 | NCBI:33213 | NCBI:33511 | NCBI:7711 | NCBI:89593 | NCBI:7742 | NCBI:7776 | NCBI:117570 | NCBI:117571 | NCBI:8287 | NCBI:1338369 | NCBI:32523 | NCBI:32524 | NCBI:40674 | NCBI:32525 | NCBI:9347 | NCBI:1437010 | NCBI:314145 | NCBI:33554 | NCBI:379584 | NCBI:9608 | NCBI:9611 | NCBI:9612 | NCBI:9615	|  | superkingdom | clade | kingdom | clade | clade | clade | phylum | subphylum | clade | clade | clade | clade | superclass | clade | clade | clade | class | clade | clade | clade | superorder | order | suborder | family | genus | species | subspecies	https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=9615	
```

**Comment by jhpoelen** [comment]
functionality was included in Nomer v0.1.25 . If needed, support matching common names can be extended beyond the lists that NCBI keeps.
